packages/arb-ts/src/lib/message/L2ToL1Message.ts

```diff
@@ -0,0 +1,374 @@
+    const outbox = IOutbox__factory.connect(this.outboxAddress, this.l1Provider)
+    return await outbox.outboxEntryExists(this.batchNumber)
+  }
+
+  public static async tryGetProof(
+    l2Provider: Provider,
+    batchNumber: BigNumber,
+    indexInBatch: BigNumber
+  ): Promise<MessageBatchProofInfo | null> {
+    const nodeInterface = NodeInterface__factory.connect(
+      NODE_INTERFACE_ADDRESS,
+      l2Provider
+    )
+    try {
+      return nodeInterface.lookupMessageBatchProof(batchNumber, indexInBatch)
+    } catch (e) {
+      const expectedError = "batch doesn't exist"
+      const err = e as any
+      const actualError =
+        err && (err.message || (err.error && err.error.message))
+      if (actualError.includes(expectedError)) {
+        console.log('Withdrawal detected, but batch not created yet.')
+      } else {
+        console.log("Withdrawal proof didn't work. Not sure why")
+        console.log(e)
+      }
+    }
+    return null
+  }
+
+  public async tryGetProof(l2Provider: Provider) {
+    return L2ToL1MessageReader.tryGetProof(
+      l2Provider,
+      this.batchNumber,
+      this.indexInBatch
+    )
+  }
+
+  /**
+   * Check if given outbox message has already been executed
+   */
+  public async hasExecuted(proofInfo: MessageBatchProofInfo): Promise<boolean> {
+    const outbox = Outbox__factory.connect(this.outboxAddress, this.l1Provider)
+    try {
+      await outbox.callStatic.executeTransaction(
+        this.batchNumber,
+        proofInfo.proof,
+        proofInfo.path,
+        proofInfo.l2Sender,
+        proofInfo.l1Dest,
+        proofInfo.l2Block,
+        proofInfo.l1Block,
+        proofInfo.timestamp,
+        proofInfo.amount,
+        proofInfo.calldataForL1
+      )
+      return false
+    } catch (err) {
+      const e = err as any
+      if (e && e.message && e.message.toString().includes('ALREADY_SPENT')) {
+        return true
+      }
+      if (e && e.message && e.message.toString().includes('NO_OUTBOX_ENTRY')) {
+        return false
+      }
+      throw e
+    }
+  }
+
+  public async status(proofInfo: MessageBatchProofInfo | null) {
+    try {
+      if (proofInfo) {
+        const messageExecuted = await this.hasExecuted(proofInfo)
+        if (messageExecuted) {
+          return OutgoingMessageState.EXECUTED
+        }
+      }
+
+      const outboxEntryExists = await this.outboxEntryExists()
+
+      return outboxEntryExists
+        ? OutgoingMessageState.CONFIRMED
+        : OutgoingMessageState.UNCONFIRMED
+    } catch (e) {
+      // CHRIS: discuss all the console logs going on in here and elsewhere, should we keep them?
+      // CHRIS: this error needs updating. also 666?
+      console.warn('666: error in getOutGoingMessageState:', e)
+      return OutgoingMessageState.NOT_FOUND
+    }
+  }
+
+  public async waitUntilOutboxEntryCreated(retryDelay = 500): Promise<void> {
+    // CHRIS: should we even be doing this? what's the use case? it could be a long wait
```

Comment:
agreed this is a weird method; no harm in keeping, but could get rid

Reply:
I got rid for now